scipy/optimize/_qap.py

```diff
@@ -0,0 +1,776 @@
+        where :math:`J` is the "barycenter" and :math:`K` is a random
+        doubly stochastic matrix.
+
+    shuffle_input : bool (default = False)
+        To avoid artificially high or low matching due to inherent
+        sorting of input matrices, gives users the option
+        to shuffle the nodes. Results are then unshuffled so that the
+        returned results correspond with the node order of inputs.
+        Shuffling may cause the algorithm to be non-deterministic,
+        unless a random seed is set or an `rng` option is provided.
+
+    maxiter : int, positive (default = 30)
+        Integer specifying the max number of Franke-Wolfe iterations performed.
+
+    tol : float (default = 0.05)
+        A threshold for the stopping criterion. Franke-Wolfe
+        iteration terminates when the change in search position between
+        iterations is sufficiently small, that is, when the Frobenius
+        norm of :math:`(P_{i}-P_{i+1}) \leq tol`, where :math:`i` is
+        the iteration number.
+
+    Returns
+    -------
+    res : OptimizeResult
+        A :class:`scipy.optimize.OptimizeResult` containing the following
+        fields.
+
+        col_ind : 1-D array
+            An array of column indices corresponding with the best
+            permutation of the nodes of `B` found.
+        fun : float
+            The corresponding value of the objective function.
+        nit : int
+            The number of Franke-Wolfe iterations performed.
+
+    Notes
+    -----
+    The algorithm may be sensitive to the initial permutation matrix (or
+    search "position") due to the possibility of several local minima
+    within the feasible region. A barycenter initialization is more likely to
+    result in a better solution than a single random initialization. However,
+    ``quadratic_assignment`` calling several times with different random
+    initializations may result in a better optimum at the cost of longer
+    total execution time.
+
+    Examples
+    --------
+    As mentioned above, a barycenter initialization often results in a better
+    solution than a single random initialization.
+
+    >>> np.random.seed(0)
+    >>> n = 15
+    >>> A = np.random.rand(n, n)
+    >>> B = np.random.rand(n, n)
+    >>> res = quadratic_assignment(A, B)  # FAQ is default method
+    >>> print(res.fun)
+    46.871483385480545 # may vary
+
+    >>> options = {"P0": "randomized"}  # use randomized initialization
+    >>> res = quadratic_assignment(A, B, options=options)
+    >>> print(res.fun)
+    47.224831071310625 # may vary
+
+    However, consider running from several randomized initializations and
+    keeping the best result.
+    >>> res = min([quadratic_assignment(A, B, options=options)
+    ...            for i in range(30)], key=lambda x: x.fun)
+    >>> print(res.fun)
+    46.671852533681516 # may vary
+
+    The '2-opt' method can be used to further refine the results.
```

Comment:
```suggestion
    The '2-opt' method can be used to further refine the results.
    
```
Same rendering issue here as above.